src/test/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowPluginSpec.groovy

```diff
@@ -1217,6 +1217,41 @@ class ShadowPluginSpec extends PluginSpecification {
         assert result.output.contains('TestApp: Hello World! (foo)')
     }
 
+    @Issue("https://github.com/GradleUp/shadow/pull/459")
+    def 'exclude gradleApi() by default'() {
+        given:
+        buildFile.text = getDefaultBuildScript('java-gradle-plugin')
+        buildFile << """
+            dependencies {
+                compileOnly gradleApi()
+            }
+        """.stripIndent()
+
+        file('src/main/java/my/plugin/MyPlugin.java') << """
+            package my.plugin;
+            import org.gradle.api.Plugin;
+            import org.gradle.api.Project;
+            public class MyPlugin implements Plugin<Project> {
+                public void apply(Project project) {
+                    System.out.println("MyPlugin: Hello World!");
+                }
+            }
+        """.stripIndent()
+        file('src/main/resources/META-INF/gradle-plugins/my.plugin.properties') << """
+            implementation-class=my.plugin.MyPlugin
+        """.stripIndent()
+
+        when:
+        run('shadowJar')
+
+        then:
+        assert output.exists()
+
+        and:
+        JarFile jar = new JarFile(output)
+        assert jar.entries().collect().findAll { it.name.endsWith('.class') }.size() == 1
```

Comment:
As you can see, Gradle APIs are included in the current jar:

https://scans.gradle.com/s/w4ng474mk6moe/tests/task/:test/details/com.github.jengelman.gradle.plugins.shadow.ShadowPluginSpec/exclude%20gradleApi()%20by%20default?top-execution=1

```
my/plugin/MyPlugin.class, 
org/gradle/process/internal/worker/GradleWorkerMain.class, 
org/gradle/process/internal/worker/WorkerJvmMemoryInfoSerializer$1.class, 
org/gradle/process/internal/worker/WorkerJvmMemoryInfoSerializer$JvmMemoryStatusSerializer.class, 
org/gradle/process/internal/worker/WorkerJvmMemoryInfoSerializer.class, 
org/gradle/process/internal/worker/WorkerLoggingSerializer.class,
…	
```